kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/resourcelock/ResourceLock.java

```diff
@@ -99,4 +97,10 @@ void setResource(T resource) {
     this.resource = resource;
   }
 
+  private static ObjectMeta validateMeta(ObjectMeta meta) {
```

Comment:
This method is not reused elsewhere and hides the intention of the null checks in the constructor (they are basically replacements for typical `@nonnull` annotations that would otherwise be added to the constructor fields)

It would be nice to know what is the main reasoning behind these isolated PRs.
Maybe we can discuss first in a regular issue to see if there is proper alignment.

Reply:
That's OK, but I think it might be better that you create an issue to encapsulate all of you findings. Then we can assess one by one if it does or does not make sense integrating those changes into the code base.